### ISSUE TITLE
raft topology: test concurrent bootstrap

### ISSUE THREAD

**Issue by patjed41** [scylladb/scylladb#15423]
In the Raft-based topology concurrent bootstraps are supported. We should test it and make use of it in the testing framework. In particular, in every test in the `broadcast_tables` and `topology` suits all starting nodes should bootstrap concurrently. We can also add a new function to `ManagerClient` that adds multiple servers concurrently.

**Comment by patjed41** [comment]
For now, I tried a simple experiment.

This test passes:
```python
servers = [await manager.server_add() for _ in range(3)]
await manager.server_stop_gracefully(servers[0].server_id)
replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = False, use_host_id = False)
await manager.server_add(replace_cfg)
```

After changing the first line to
```python
await asyncio.gather(*(manager.server_add() for _ in range(3)))
servers = await manager.running_servers()
```
it doesn't pass. The replacing node fails with:
```
ERROR 2023-09-14 16:10:28,146 [shard 0:main] init - Startup failed: std::runtime_error (Cannot replace_address 127.199.3.2 because it doesn't exist in gossip)
```

I'll try to investigate this.

**Comment by patjed41** [comment]
It turnes out that all concurrently bootstrapping servers set themselves as the only seed and basically start a new cluster. In `ScyllaCluster.add_server` we have:
```python
seeds = self._seeds()
if not seeds:
    seeds = [ip_addr]
```
So, if the cluster is empty, a new `ScyllaCluster.add_servers` function would have to somehow make sure that all starting servers use IP of a single chosen server as the only seed.

**Comment by kbr-scylla** [comment]
Ahh, so simple concurrent `add_server()` calls won't do, we need something more clever...

> would have to somehow make sure that all starting servers use IP of a single chosen server as the only seed

Another way is to set all `n` servers as seeds on every server. We should support such configuration too.

**Comment by patjed41** [comment]
I opened #15435 after detecting a bug. It blocks this issue.

**Comment by patjed41** [comment]
This is not fixed. I reopened.

It looks like it was closed because of an unfortunate match in the description of  #16102 -- "This PR is the first step to fix #15423."
